### DISCUSSION TITLE
Tp-link Wifi

### DISCUSSION THREAD

**Discussion by st623** [SpiralLinux/SpiralLinux-project#171] [Q&A]
Sorry for my English.
I tried the live-usb, KDE edition.
The TP-Link WN321g worked out of the box and with full signal (on Kubuntu and OpenSuse it works with poor signal due to the drivers)
But the newer TP-Link WN821N v6.0 is not detected, doesn't show in the connections or blink the light. It worked on Kubuntu and OpenSuse (although with poor signal; it appears youneed to install the proprietary driver on those distros)

**Comment by geckolinux** [comment]
Hi there, thanks for the report. Could you please try the TP-Link WN821N with both included kernels? (5.10 and 5.18).

**Comment by st623** [comment]
I tried both kernels yesterday but it didn't  work
The chipset is RTL8192EU; i tried to view the installed packages and it appears this driver is installed by default, so I don't know why it doesn't detect the dongle
I would test on GeckoLinux too, but Gox is offline (https://gox.dl.sourceforge.net/project/geckolinux/Static/154.220822/GeckoLinux_STATIC_Plasma.x86_64-154.220822.0.iso). I will try again at night.

**Reply by geckolinux** [reply]
To download from Sourceforge you can click on the "Problems downloading?" link and select a different mirror.

From what I can tell the dongle requires a kernel driver in addition to the firmware, and Debian does not offer any package for the driver. I'm not sure if this is only required for the USB dongle version or if internal chips would also be troublesome. I try to include as much support for Realtek WiFI out-of-the-box as Debian provides, but in general you're in for a bad time with Realtek WiFI under Linux (and even under Windows, from what I've read.) At any rate it looks like unfortunately you'll have to compile the driver:

- https://unix.stackexchange.com/questions/705689/debian-buster-package-firmware-realtek-installed-rtl8192eu-wifi-dongle-usb-ada
- https://wiki.debian.org/WiFi#Availability_of_compatible_WiFi_chipsets
- https://forum.mxlinux.org/viewtopic.php?t=59382

It looks like this fork is the most up-to-date and easiest to install:
https://github.com/clnhub/rtl8192eu-linux#automated-install

**Reply by st623** [reply]
Thanks
